### PR TITLE
Add rspec to workflows

### PR DESCRIPTION
**Pull request by enderahmetyurt**

### CODE REVIEW

**Comment by enderahmetyurt** [comment]
@kiliczsh mate would you like to help me here 🙏

**Comment by kiliczsh** [comment]
Meanwhile can you try to create a database for test environment. The command output is ActiveRecord::ConnectionNotEstablished as can be seen below.


```bash
An error occurred while loading rails_helper.
Failure/Error: ActiveRecord::Migration.maintain_test_schema!

ActiveRecord::ConnectionNotEstablished:
  connection to server at "::1", port 5432 failed: Connection refused
  	Is the server running on that host and accepting TCP/IP connections?
  connection to server at "[12](https://github.com/enderahmetyurt/podiscover/actions/runs/5444793848/jobs/9903130498?pr=159#step:8:13)7.0.0.1", port 5432 failed: Connection refused
  	Is the server running on that host and accepting TCP/IP connections?
# ./spec/rails_helper.rb:30:in `<top (required)>'
# ------------------
# --- Caused by: ---
# PG::ConnectionBad:
#   connection to server at "::1", port 5432 failed: Connection refused
#   	Is the server running on that host and accepting TCP/IP connections?
#   connection to server at "127.0.0.1", port 5432 failed: Connection refused
#   	Is the server running on that host and accepting TCP/IP connections?
#   ./spec/rails_helper.rb:30:in `<top (required)>'
No examples found.
```

**Comment by kiliczsh** [comment]
Taking notes as I am running 

```bash
RAILS_ENV=test DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL=true bundle exec rspec --require rails_helper
```
Screenshot 2023-07-07 at 17 50 04


We may need to add this environment variable
```env
DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL=true
```
due to this error: 
```ruby
DatabaseCleaner::Safeguard::Error::RemoteDatabaseUrl: ENV['DATABASE_URL'] is set to a remote URL. 
Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards
```

**Comment by kiliczsh** [comment]
```bash
Pending: (Failures listed here are expected and do not affect your suite's status)
...
 .. add some examples to (or delete)
 # Not yet implemented
 # ./spec/models/followable_spec.rb:4
 # ./spec/models/likeable_spec.rb:4
 # ./spec/models/openai_request_spec.rb:4
 # ./spec/models/user_listened_episode_spec.rb:4

Finished in 9 minutes 28 seconds (files took 12.96 seconds to load)
41 examples, 0 failures, 4 pending
```

**Comment by kiliczsh** [comment]
@enderahmetyurt we need environment secrets for the list below:
```env
POSTGRES_USER: ${{ secrets.TEST_POSTGRES_USER }}
POSTGRES_PASSWORD: ${{ secrets.TEST_POSTGRES_PASSWORD }}
POSTGRES_DB: ${{ secrets.TEST_POSTGRES_DB }}
POSTGRES_HOST: ${{ secrets.TEST_POSTGRES_HOST }}
POSTGRES_PORT: ${{ secrets.TEST_POSTGRES_PORT }}
DATABASE_URL: ${{ secrets.TEST_DATABASE_URL }}
```

File: db/schema.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@
 ActiveRecord::Schema[7.0].define(version: 2023_06_17_075921) do
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
+  enable_extension "timescaledb"
```
Comment (enderahmetyurt) [inline db/schema.rb:16]: why do we need this? @kiliczsh

Comment (kiliczsh) [reply db/schema.rb:16]: I have integrated with Railway they are using timescale db
will it create a performance issue?

**Comment by kiliczsh** [comment]
database_cleaner gem is missing and 
DATABASE_URL is empty afair from the error logs

**Comment by kiliczsh** [comment]
@enderahmetyurt I think this is what you wanted right?

It fails right now due to RSpec failures: 

```bash
Run COVERAGE=true bundle exec rspec --require rails_helper
.F....***.*FFFFFFFFFFFFFFFFFFFFFFFFFFF...

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Followable add some examples to (or delete) /home/runner/work/podiscover/podiscover/spec/models/followable_spec.rb
     # Not yet implemented
     # ./spec/models/followable_spec.rb:4

  2) Likeable add some examples to (or delete) /home/runner/work/podiscover/podiscover/spec/models/likeable_spec.rb
     # Not yet implemented
     # ./spec/models/likeable_spec.rb:4

  3) OpenaiRequest add some examples to (or delete) /home/runner/work/podiscover/podiscover/spec/models/openai_request_spec.rb
     # Not yet implemented
     # ./spec/models/openai_request_spec.rb:4

  4) UserListenedEpisode add some examples to (or delete) /home/runner/work/podiscover/podiscover/spec/models/user_listened_episode_spec.rb
     # Not yet implemented
     # ./spec/models/user_listened_episode_spec.rb:4

Failures:

  1) FetchUserPodcasts should fetch a given user podcasts and their images
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/jobs/fetch_user_podcasts_spec.rb:28:in `block (2 levels) in <main>'

  2) User after create callbacks should send new user email after a user created
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:5:in `block (3 levels) in <main>'
     # ./spec/models/user_spec.rb:8:in `block (3 levels) in <main>'

  3) User after create callbacks should create related broadcastable
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:5:in `block (3 levels) in <main>'
     # ./spec/models/user_spec.rb:13:in `block (4 levels) in <main>'
     # ./spec/models/user_spec.rb:13:in `block (3 levels) in <main>'

  4) User slug vales should be parameterized
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:20:in `block (3 levels) in <main>'

  5) User slug vales should have appended 4 character uid if nickname is used as a slug for a different user
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:20:in `block (3 levels) in <main>'

  6) User#activity_feed should show new user's new_user_has_joined broadcastable activity
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:37:in `block (3 levels) in <main>'

  7) User#activity_feed should not show to user its own new_user_has_joined broadcastable activity
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:37:in `block (3 levels) in <main>'

  8) User#activity_feed should show followed user's likeable activity
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:37:in `block (3 levels) in <main>'

  9) User#activity_feed should create a followed activity when user starts to follow another user
     Failure/Error: UserMailer.new_user(self).deliver_now

     ArgumentError:
       SMTP To address may not be blank: []
     # ./app/models/user.rb:115:in `send_new_user_email'
     # ./spec/models/user_spec.rb:37:in `block (3 levels) in <main>'

  [10](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:11)) User#activity_feed should create an unfollowed activity when user stops to follow another user
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:[11](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:12)5:in `send_new_user_email'
      # ./spec/models/user_spec.rb:37:in `block (3 levels) in <main>'

  11) Podcast#show returns 200 when a podcast exists
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/podcasts_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/podcasts_request_spec.rb:6:in `block (2 levels) in <main>'

  [12](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:13)) Podcast#show returns 302 when a podcast does not exists
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/podcasts_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/podcasts_request_spec.rb:6:in `block (2 levels) in <main>'

  [13](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:14)) Podcast#search returns podcast by query
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/podcasts_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/podcasts_request_spec.rb:6:in `block (2 levels) in <main>'

  [14](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:15)) Podcast#search show no podcast found message if any podcast not found by query
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:1[15](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:16):in `send_new_user_email'
      # ./spec/requests/podcasts_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/podcasts_request_spec.rb:6:in `block (2 levels) in <main>'

  15) Podcast#search shows all podcasts without query
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/podcasts_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/podcasts_request_spec.rb:6:in `block (2 levels) in <main>'

  [16](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:17)) Episode#index 
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [17](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:18)) Episode#show supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [18](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:19)) Episode#show supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [19](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:20)) Episode#following supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [20](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:21)) Episode#following supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [21](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:22)) Episode#followers supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [22](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:23)) Episode#followers supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [23](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:24)) Episode#email_confirmation supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [24](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:25)) Episode#email_confirmation supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [25](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:26)) Episode#allow_email_confirmation supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [26](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:27)) Episode#allow_email_confirmation supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [27](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:28)) Episode#deny_email_confirmation supports :id
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

  [28](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:29)) Episode#deny_email_confirmation supports :slug
      Failure/Error: UserMailer.new_user(self).deliver_now

      ArgumentError:
        SMTP To address may not be blank: []
      # ./app/models/user.rb:115:in `send_new_user_email'
      # ./spec/requests/users_request_spec.rb:4:in `block (2 levels) in <main>'
      # ./spec/requests/users_request_spec.rb:6:in `block (2 levels) in <main>'

Finished in 0.53752 seconds (files took 3.11 seconds to load)
41 examples, 28 failures, 4 pending

Failed examples:

rspec ./spec/jobs/fetch_user_podcasts_spec.rb:6 # FetchUserPodcasts should fetch a given user podcasts and their images
rspec ./spec/models/user_spec.rb:7 # User after create callbacks should send new user email after a user created
rspec ./spec/models/user_spec.rb:12 # User after create callbacks should create related broadcastable
rspec ./spec/models/user_spec.rb:23 # User slug vales should be parameterized
rspec ./spec/models/user_spec.rb:27 # User slug vales should have appended 4 character uid if nickname is used as a slug for a different user
rspec ./spec/models/user_spec.rb:43 # User#activity_feed should show new user's new_user_has_joined broadcastable activity
rspec ./spec/models/user_spec.rb:50 # User#activity_feed should not show to user its own new_user_has_joined broadcastable activity
rspec ./spec/models/user_spec.rb:57 # User#activity_feed should show followed user's likeable activity
rspec ./spec/models/user_spec.rb:61 # User#activity_feed should create a followed activity when user starts to follow another user
rspec ./spec/models/user_spec.rb:65 # User#activity_feed should create an unfollowed activity when user stops to follow another user
rspec ./spec/requests/podcasts_request_spec.rb:9 # Podcast#show returns 200 when a podcast exists
rspec ./spec/requests/podcasts_request_spec.rb:18 # Podcast#show returns [30](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:31)2 when a podcast does not exists
rspec ./spec/requests/podcasts_request_spec.rb:26 # Podcast#search returns podcast by query
rspec ./spec/requests/podcasts_request_spec.rb:35 # Podcast#search show no podcast found message if any podcast not found by query
rspec ./spec/requests/podcasts_request_spec.rb:42 # Podcast#search shows all podcasts without query
rspec ./spec/requests/users_request_spec.rb:9 # Episode#index 
rspec ./spec/requests/users_request_spec.rb:18 # Episode#show supports :id
rspec ./spec/requests/users_request_spec.rb:25 # Episode#show supports :slug
rspec ./spec/requests/users_request_spec.rb:[34](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:35) # Episode#following supports :id
rspec ./spec/requests/users_request_spec.rb:[41](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:42) # Episode#following supports :slug
rspec ./spec/requests/users_request_spec.rb:[50](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:51) # Episode#followers supports :id
rspec ./spec/requests/users_request_spec.rb:[57](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:58) # Episode#followers supports :slug
rspec ./spec/requests/users_request_spec.rb:[66](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:67) # Episode#email_confirmation supports :id
rspec ./spec/requests/users_request_spec.rb:[74](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:75) # Episode#email_confirmation supports :slug
rspec ./spec/requests/users_request_spec.rb:[84](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:85) # Episode#allow_email_confirmation supports :id
rspec ./spec/requests/users_request_spec.rb:[93](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:94) # Episode#allow_email_confirmation supports :slug
rspec ./spec/requests/users_request_spec.rb:[104](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:105) # Episode#deny_email_confirmation supports :id
rspec ./spec/requests/users_request_spec.rb:[113](https://github.com/enderahmetyurt/podiscover/actions/runs/5488915819/jobs/10002402615#step:8:114) # Episode#deny_email_confirmation supports :slug

Error: Process completed with exit code 1.
```

File: db/schema.rb (hunk 1 above)
Comment (kiliczsh) [reply db/schema.rb]: Removed timescaledb extension

**Comment by enderahmetyurt** [comment]
> 

Thank you @kiliczsh these tests are not failing in my local. Could you try it on yours? I suppose this error appears since some smtp settings or something similiar hasn't done in workflow.

Screenshot 2023-07-12 at 09 49 23

**Comment by kiliczsh** [comment]
> > 
> 
> Thank you @kiliczsh these tests are not failing in my local. Could you try it on yours? I suppose this error appears since some smtp settings or something similiar hasn't done in workflow.
> 
> Screenshot 2023-07-12 at 09 49 23

It shows 4 pending for me also. @enderahmetyurt I checked and saw ADMIN_EMAIL environment variable can you add that as environment variable not secret for the repository then we will see if we need further improvements

**Comment by enderahmetyurt** [comment]
@kiliczsh thanks man. this error is gone but another one shows up :)

**Comment by kiliczsh** [comment]
@enderahmetyurt There are two errors:
1. is redis connection problem, will take a look for this after 2.
2. is asset compilation error, `bundle exec rails assets:precompile` is the correct command right?

**Comment by kiliczsh** [comment]
@enderahmetyurt we are almost ready to merge 🚀 :)

**Comment by enderahmetyurt** [comment]
THANK YOU VERY MUCH MAN. You're the KING! 👑

**Comment by kiliczsh** [comment]
> @kiliczsh we're getting an error. I didn't add this to the codebase. Would I have done it?

@enderahmetyurt I couldn't see the errors? These fields are either from .env file/environment variables or from repository secrets

**Comment by enderahmetyurt** [comment]
> > @kiliczsh we're getting an error. I didn't add this to the codebase. Would I have done it?
> 
> @enderahmetyurt I couldn't see the errors? These fields are either from .env file/environment variables or from repository secrets

could you look PR's? All PR's in failed status now because of env variable thing I suppose

**Comment by enderahmetyurt** [comment]
secrets in repo are there otherwise commits in main brach would be failed, right?

**Pull request merged**